### PR TITLE
Use pagination when listing large numbers of resources

### PR DESCRIPTION
**Pull request by brandond**
#### Proposed Changes ####

Use pagination when listing large numbers of resources

etcd snapshots and secrets are the only thing we bulk list, everything else should either have a selector, or there aren't enough resource of that type to matter.

#### Types of Changes ####

enhancement

#### Verification ####

See linked issue - note that this requires a LARGE number of snapshots to make a visible difference in memory - at least 100

#### Linked Issues ####

* https://github.com/k3s-io/k3s/issues/10450

#### User-Facing Change ####
```release-note
```

### CODE REVIEW

File: pkg/secretsencrypt/controller.go [hunk 1]
```diff
@@ -30,6 +29,8 @@ const (
 	secretsProgressEvent       string = "SecretsProgress"
 	secretsUpdateCompleteEvent string = "SecretsUpdateComplete"
 	secretsUpdateErrorEvent    string = "SecretsUpdateError"
+
+	secretListPageSize = 20
```
Comment (dereknola) [inline pkg/secretsencrypt/controller.go:33]: Why such a small page size? the default is 500

File: pkg/secretsencrypt/controller.go [hunk 2]
```diff
@@ -213,36 +214,29 @@ func (h *handler) validateReencryptStage(node *corev1.Node, annotation string) (
...
-				return fmt.Errorf("failed to update secret: %v", err)
-			}
-			if i != 0 && i%10 == 0 {
-				h.recorder.Eventf(nodeRef, corev1.EventTypeNormal, secretsProgressEvent, "reencrypted %d secrets", i)
-			}
-			i++
+	if err := secretPager.EachListItem(h.ctx, metav1.ListOptions{}, func(obj runtime.Object) error {
+		secret, ok := obj.(*corev1.Secret)
+		if !ok {
+		}
```
Comment (dereknola) [inline pkg/secretsencrypt/controller.go:227]: Empty error check

**Review by dereknola — CHANGES_REQUESTED** [review]

File: pkg/secretsencrypt/controller.go (hunk 1 above)
Comment (brandond) [reply pkg/secretsencrypt/controller.go:34]: This is tuned base on the resource utilization graphs at https://github.com/k3s-io/k3s/issues/10450. That issue doesn't have screenshots of the secrets-encrypt stuff specifically, but you can see the result of listing all snapshots with the default page size - it takes a TON of memory for the apiserver to collect all the objects, deserialize them from the storage format, convert and serialize them back into a result response body, send them to the client, have the client deserialize them again, and store them in memory until the function is done processing them. When etcd and the apiserver all run in separate processes the memory is spread out a bit more, but here in K3s we get dinged for every step of that being in the same process. We can keep the peak memory utilization much lower if we limit the page size.

File: pkg/secretsencrypt/controller.go (hunk 2 above)
Comment (brandond) [reply pkg/secretsencrypt/controller.go]: fixed!

**Review by dereknola — APPROVED** [review]

**Review by vitorsavian — APPROVED** [review]

**Pull request merged**